src/wallet/walletdb.cpp

```diff
@@ -216,6 +217,35 @@ bool CWalletDB::EraseSaplingExtendedFullViewingKey(
     return Erase(std::make_pair(std::string("sapextfvk"), extfvk));
 }
 
+//
+// Unified address & key storage
+//
+
+bool CWalletDB::WriteUnifiedSpendingKeyMetadata(const ZcashdUnifiedSpendingKeyMetadata& keymeta)
+{
+    nWalletDBUpdateCounter++;
+    auto ufvkId = keymeta.GetKeyID();
+    return Write(std::make_pair(std::string("unifiedskmeta"), ufvkId), keymeta);
+}
+
+bool CWalletDB::WriteUnifiedFullViewingKey(const libzcash::UnifiedFullViewingKey& ufvk)
+{
+    nWalletDBUpdateCounter++;
+    auto ufvkId = ufvk.GetKeyID(Params());
+    return Write(std::make_pair(std::string("unifiedfvk"), ufvkId), ufvk.Encode(Params()));
+}
+
+bool CWalletDB::WriteUnifiedAddressMetadata(const ZcashdUnifiedAddressMetadata& addrmeta)
+{
+    nWalletDBUpdateCounter++;
+    auto ufvkId = addrmeta.GetKeyID();
+    return Write(std::make_pair(std::string("unifiedaddrmeta"), ufvkId), addrmeta);
```

Comment:
I think that what I intended to do here is basically what you suggest, but instead of storing `addrmeta` as the value just store the set of receivers. I already reconstruct the `ufvkid` part of the ZcashdUnifiedAddressMetadata` from the key on read.

Reply:
We don't have native serialization for `std::set` though, which is why I was just storing the whole ZUAM.